### PR TITLE
Add documentation for the LEDControl plugin

### PR DESCRIPTION
**Pull request by EvyBongers**

### CODE REVIEW

File: plugins/Kaleidoscope-LEDControl/README.md [hunk 1]
```diff
@@ -1,6 +1,118 @@
...
+
+### `.getCrgbAt(uint8_t led_index)`
+
+> Get the LED color of the specified LED.
+
+### `.getCrgbAt(KeyAddr key_addr)`
+
+> Get the LED color of the LED for the specified key.
+
+### `.syncLeds(void)`
```
Comment (EvyBongers) [inline plugins/Kaleidoscope-LEDControl/README.md:55]: This one seems pretty straight forward to me, but I failed to come up with a good way to phrase this.

File: plugins/Kaleidoscope-LEDControl/README.md [hunk 2]
```diff
@@ -1,6 +1,118 @@
...
+
+> Activates the next LED mode. Cycles to the first LED mode if the current LED
+> mode is the last one.
+
+### `.prev_mode(void)`
+
+> Activates the previous LED mode. Cycles to the last LED mode if the current
+> LED mode is the first one.
+
+### `.get_mode()`
```
Comment (EvyBongers) [inline plugins/Kaleidoscope-LEDControl/README.md:22]: There are two overloads for this method and I'm yet to understand what one of them does. Asked the question on Discord, but leaving this here as a reminder.

File: plugins/Kaleidoscope-LEDControl/README.md [hunk 3]
```diff
@@ -1,6 +1,118 @@
...
+
+### `.getBrightness()`
+
+> Returns the current brightness of the LEDs as a uint8_t.
+
+### `.onSetup()`
+
+> See [[event-handler-hooks]]
+
+### `.setup(void)`
```
Comment (EvyBongers) [inline plugins/Kaleidoscope-LEDControl/README.md:86]: I'm not sure how to explain this one. I half expect it to be documented elsewhere and that I should refer to that, but I failed to find it.

File: plugins/Kaleidoscope-LEDControl/README.md [hunk 4]
```diff
@@ -1,6 +1,118 @@
...
+
+> Triggers the currently active LED mode to update. It is up to the LED mode to
+> handle this correctly.
+
+### `.refreshAt(KeyAddr key_addr)`
+
+> Triggers the currently active LED mode to refresh the LED at the specified key
+> address.
+
+### `.activate(LEDModeInterface *plugin)`
```
Comment (EvyBongers) [inline plugins/Kaleidoscope-LEDControl/README.md:105]: I was about to write something along the lines of "Activates the specified LED mode." However, that would not prevent the judgement error recently pointed out to me that this actually invokes and depends on PROGMEM and/or EEPROM.

**Comment by obra** [comment]
Thank you so much for working on this. I do so wish that we had a decent way for method-level docs to live next to the methods they're documenting and to generate pretty looking readable docs.

**Comment by EvyBongers** [comment]
Doxygen can't do that?

https://www.doxygen.nl/manual/docblocks.html

File: plugins/Kaleidoscope-LEDControl/README.md [hunk 5]
```diff
@@ -1,6 +1,123 @@
...
+### `.prev_mode(void)`
+
+> Activates the previous LED mode. Cycles to the last LED mode if the current
+> LED mode is the first one.
+
+### `.get_mode()`
+
+> Returns the current LED mode.
+
+### `.get_mode<typename>()`
```
Comment (EvyBongers) [inline plugins/Kaleidoscope-LEDControl/README.md:26]: I don't know how to properly explain this. Having looked at [the commit](https://github.com/keyboardio/Kaleidoscope/commit/f54e11a9b2604ec0fe2931549773bbec42815397) you mentioned on discord I understand what it does, but I don't know the words to explain it.

**Comment by EvyBongers** [comment]
This is the best I can do at this time

**Comment by obra** [comment]
This is great. Thanks!

**Pull request merged**